### PR TITLE
feat: unread conversation and message indicator (AR-918)

### PR DESCRIPTION
**Pull request by trOnk12**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

-This is the final PR merge to the develop of : add unread indicator feature support

### Solutions

- Add last_read_date on a Conversation TABLE 
- Add SQL queries to : get amount conversations that have unread messages by comparing last_modified_date with last_read_date , get amount of messages that are unread for a conversation by comparing date with last_read_date 
- Expose the unread count by adding unreadConversationsCount to a UseCase return object 
logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/ObserveConversationListDetailsUseCaseImpl.kt

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq [hunk 1]
```diff
@@ -15,6 +15,7 @@ CREATE TABLE Conversation (
     muted_time INTEGER DEFAULT 0 NOT NULL,
     last_modified_date TEXT NOT NULL,
     last_notified_message_date TEXT,
+    last_read_date TEXT,
```
Comment (typfel) [inline persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq:18]: Not for this PR but:
I think we should consider storing `last_read_date` (and all other timestamps which are always in the past) as an integer.

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAO.kt [hunk 2]
```diff
@@ -12,6 +12,9 @@ data class ConversationEntity(
     val mutedTime: Long = 0,
     val lastNotificationDate: String?,
     val lastModifiedDate: String,
+    // Date that indicates when the user has seen the conversation,
+    // nullable lastReadDate means that the conversation is never seen by the user
+    val lastReadDate: String? = null,
```
Comment (typfel) [inline persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAO.kt:17]: Regardless of how we store this in SQLite we can convert it to an Instant at this level and above. If this is a big change I'm fine with doing in a separate PR.

**Review by typfel — CHANGES_REQUESTED** [review]
Looking good! 💯 Just some typos and comments for future improvement which might not belong to this PR.

File: persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAO.kt (hunk 2 above)
Comment (trOnk12) [reply persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/ConversationDAO.kt:17]: I think this was discussed in last collective and there is a refactor ticket pending, i'll leave it as it for now but add a todo

File: persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq (hunk 1 above)
Comment (trOnk12) [reply persistence/src/commonMain/db_user/com/wire/kalium/persistence/Conversations.sq:18]: will add a TODO

**Review by typfel — APPROVED** [review]

**Pull request merged**